catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -929,10 +946,18 @@ private CswRecordCollection queryCsw(GetRecordsType request) throws CswException
             try {
                 queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                         request.getOutputSchema());
-                LOGGER.debug("Attempting to execute query: {}", queryRequest);
-                QueryResponse queryResponse = framework.query(queryRequest);
+                queryRequest = setQueryPageSize(queryRequest);
+
+                LOGGER.debug("Attempting to execute paged query: {}", queryRequest);
+                ResultIterable results = new ResultIterable(framework, queryRequest);
+                List<Result> resultList = results.stream()
+                        .collect(Collectors.toList());
+
+                QueryResponse queryResponse = new QueryResponseImpl(queryRequest,
+                        resultList,
+                        resultList.size());
                 response.setSourceResponse(queryResponse);
-            } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            } catch (UnsupportedQueryException e) {
```

Comment:
❓ Shouldn't we catch `QueryResultException` here and pass the cause as a parameter to the `CswException`?